compiler/rustc_trait_selection/src/error_reporting/infer/need_type_info.rs

```diff
@@ -628,6 +805,7 @@ enum InferSourceKind<'tcx> {
         pattern_name: Option<Ident>,
         ty: Ty<'tcx>,
         def_id: Option<DefId>,
+        hir_id: Option<HirId>,
```

Comment:
Is this different than just the `tcx.local_def_id_to_hir_id(def_id)`? If so, then it should probably be given a better name.